### PR TITLE
hevm: skip unit test contracts with no code

### LINKED ISSUE #769: Do not run tests in abstract contracts
Writing `abstract contract MyTest is DSTest {` derails the testing suite:

```
Running 1 tests for src/test/MyTest.t.sol:MyTest
hevm: Data.Binary.Get.runGet at position 0: not enough bytes
uint256
bool
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:351:5 in binary-0.8.8.0:Data.Binary.Get
```

AFAIK there's no way to tell the testing suite to ignore a contract. It's included for running even before attempted compilation based on the ABI and there's no way to hide `IS_TEST`.

### PR DESCRIPTION
**Pull request by d-xo**
## Description

Skips execution for unit test contracts with no code. Fixes #769.

## Checklist

- [ ] tested locally
- [x] added automated tests
- [ ] updated the docs
- [x] updated the changelog

### CODE REVIEW

File: src/dapp-tests/pass/abstract.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.6.7;
+
+import {DSTest} from "ds-test/test.sol";
+
+// should not be run (no code)
+abstract contract MyTest is DSTest {}
```
Comment (CodeSandwich) [inline src/dapp-tests/pass/abstract.sol:6]:
```suggestion
abstract contract MyTest is DSTest {
    function abstractTest() public {
        assertTrue(true);
    }
}
```
Could we add an abstract test? This is exactly what I needed abstract tests for.

File: src/hevm/src/EVM/Dapp.hs [hunk 2]
```diff
@@ -140,7 +140,9 @@ findUnitTests match =
       Nothing -> []
       Just _  ->
         let testNames = unitTestMethodsFiltered (regexMatches match) c
-        in [(view contractName c, testNames) | not (null testNames)]
+        in if BS.null (view runtimeCode c)
+           then []
+           else [(view contractName c, testNames) | not (null testNames)]
```
Comment (MrChico) [inline src/hevm/src/EVM/Dapp.hs:145]:
```suggestion
        in [(view contractName c, testNames) | not (BS.null (view runtimeCode c) && not (null testNames)]
```

Comment (MrChico) [reply src/hevm/src/EVM/Dapp.hs:145]: I think something like this should work, and look cleaner immo

File: src/dapp-tests/pass/abstract.sol (hunk 1 above)
Comment (d-xo) [inline src/dapp-tests/pass/abstract.sol:6]:
```suggestion
abstract contract MyTest is DSTest {
    function testAbstract() public {
        assertTrue(true);
    }
}
```

**Pull request merged**